server/game/playattachmentaction.js

```diff
@@ -40,7 +40,8 @@ class PlayAttachmentAction extends BaseAction {
             originalLocation: context.source.location,
             playType: PlayTypes.PlayFromHand
         });
-        context.game.openEventWindow([context.game.actions.attach({ attachment: context.source }).getEvent(context.target, context), cardPlayedEvent]);
+        let takeControl = context.source.controller !== context.player;
```

Comment:
This looks wrong to me.  I think `takeControl` should always be `true`.  If I play an attachment using Infiltrator, I should control it

Reply:
The problem here is that if I set this to `true` in all cases then it interferes with Karada/CiF as they are explicitly checking that you can only "take control" if you are not the current controller.  This will need a bit of an overhaul I think but for now the above still works with Infiltrator as the controller of the card is not the player anyway.  I'll push a test for that though as it is one to watch out for in the future.